tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -119,7 +120,928 @@ void test_duration_output() {
+    fail_parse("2017-54-0", "%Y-%W-%w", ymd);
+    fail_parse("2017-54-0", "%Y-%U-%w", ymd);
+    fail_parse("2018-00-0", "%Y-%U-%w", ymd); // refers to 31 Dec. 2017
+    fail_parse("2017-53-1", "%Y-%U-%w", ymd); // refers to 01 Jan. 218
+
+    // Year begins on Monday.
+    test_parse("2018-00-1", "%Y-%U-%w", ymd);
+    assert(ymd == 2018y / January / 1d);
+    test_parse("2018-01-1", "%Y-%W-%w", ymd);
+    assert(ymd == 2018y / January / 1d);
+    test_parse("2018-53-1", "%Y-%W-%w", ymd);
+    assert(ymd == 2018y / December / 31d);
+
+    // Year begins on Tuesday.
+    test_parse("2019-00-2", "%Y-%U-%w", ymd);
+    assert(ymd == 2019y / January / 1d);
+    test_parse("2019-00-2", "%Y-%W-%w", ymd);
+    assert(ymd == 2019y / January / 1d);
+
+    // Year begins on Wednesday.
+    test_parse("2020-00-3", "%Y-%U-%w", ymd);
+    assert(ymd == 2020y / January / 1d);
+    test_parse("2020-00-3", "%Y-%W-%w", ymd);
+    assert(ymd == 2020y / January / 1d);
+
+    // Year begins on Thursday.
+    test_parse("2015-00-4", "%Y-%U-%w", ymd);
+    assert(ymd == 2015y / January / 1d);
+    test_parse("2015-00-4", "%Y-%W-%w", ymd);
+    assert(ymd == 2015y / January / 1d);
+
+    // Year begins on Friday.
+    test_parse("2016-00-5", "%Y-%U-%w", ymd);
+    assert(ymd == 2016y / January / 1d);
+    test_parse("2016-00-5", "%Y-%W-%w", ymd);
+    assert(ymd == 2016y / January / 1d);
+
+    // Year begins on Saturday.
+    test_parse("2022-00-6", "%Y-%U-%w", ymd);
+    assert(ymd == 2022y / January / 1d);
+    test_parse("2022-00-6", "%Y-%W-%w", ymd);
+    assert(ymd == 2022y / January / 1d);
+}
+
+void parse_whitespace() {
+    seconds time;
+    fail_parse("ab", "a%nb", time);
+    test_parse("a b", "a%nb", time);
+    fail_parse("a  b", "a%nb", time);
+    fail_parse("a", "a%n", time);
+    test_parse("a ", "a%n", time);
+
+    test_parse("ab", "a%tb", time);
+    test_parse("a b", "a%tb", time);
+    fail_parse("a  b", "a%tb", time);
+    test_parse("a", "a%t", time);
+    test_parse("a ", "a%t", time);
+
+    test_parse("a", "a ", time);
+    test_parse("", "", time);
+    test_parse("", " ", time);
+    test_parse("", "%t", time);
+    fail_parse("", "%n", time);
+}
+
+void insert_leap_second(const sys_days& date, const seconds& value) {
+    const auto& my_tzdb = get_tzdb_list().front();
+    vector<time_zone> zones;
+    vector<time_zone_link> links;
+    transform(my_tzdb.zones.begin(), my_tzdb.zones.end(), back_inserter(zones),
+        [](const auto& _Tz) { return time_zone{_Tz.name()}; });
+    transform(my_tzdb.links.begin(), my_tzdb.links.end(), back_inserter(links), [](const auto& link) {
+        return time_zone_link{link.name(), link.target()};
+    });
+
+    auto leap_vec = my_tzdb.leap_seconds;
+    leap_vec.emplace_back(date, value == 1s, leap_vec.back()._Elapsed());
+    get_tzdb_list()._Emplace_front(
+        tzdb{my_tzdb.version, move(zones), move(links), move(leap_vec), my_tzdb._All_ls_positive && (value == 1s)});

```

Comment:
I don't think it would hurt, but it wouldn't address the issue of duplicate version strings, because I'm calling this function twice. I'd have to rework the function to take all the leap seconds in a single call, or add a static counter to the version string to make it unique. The counter would be easy, but otherwise I'm not sure it's worth the effort.

Reply:
Thanks - as none of the test code is affected by the version, and this doesn't impact any product code logic, I think we can resolve this as "nothing to do".